src/utils/hooks/useSearch.ts

```diff
@@ -0,0 +1,60 @@
+import { useState, useEffect } from 'react'
+import { useSearchParams } from 'react-router-dom'
+import Featured from '../../interfaces/Featured'
+import { API_BASE_URL } from '../constants'
+import { useLatestAPI } from './useLatestAPI'
+
+export function useSearch () {
```

Comment:
all the hook that use the API have a lot of duplicated code, the could be hook that contains all that is common to these hooks, I think that the only thing that change is the URL, the hook could look something like:


```jsx
import { useState, useEffect } from "react";
import { API_BASE_URL } from "../constants";
import { useLatestAPI } from "./useLatestAPI";

export function useFetch(endpoint) {
  const { ref: apiRef, isLoading: isApiMetadataLoading } = useLatestAPI();
  const [data, setData] = useState();
  const [error, setError] = useState();
  const [loading, setLoading] = useState();

  useEffect(() => {
    if (!apiRef || isApiMetadataLoading) {
      return;
    }

    const controller = new AbortController();

    async function getData() {
      try {
        setData(null);
        setError(null);

        const response = await fetch(`${API_BASE_URL}/${endpoint}}`, {
          signal: controller.signal,
        });
        const data = await response.json();

        setData(data);
      } catch (err) {
        setError(err);
        console.error(err);
      } finally {
        setLoading(false);
      }
    }

    getData();

    return () => {
      controller.abort();
    };
  }, [apiRef, isApiMetadataLoading, endpoint]);

  return { data, error, loading };
}
```